### ISSUE TITLE
[BUG] proxy setting have no practical effect

### ISSUE THREAD

**Issue by Emphasia** [johanneszab/TumblThree#204]
blog url: https://thereisnonamehere.tumblr.com/(safe-mode) & https://action.tumblr.com/(normal) & wallpaperfx.tumblr.com(normal)
PROXY=[XX-Net](https://github.com/XX-net/XX-Net)(a HTTP proxy tool)@127.0.0.1:8087
set IE's proxy as PROXY, set TumblThree's proxy as null: can view all blogs just fine (in IE) and have connect record in PROXY'LOG
unset IE's proxy: can't connect tumblr.com (in IE)
unset IE's proxy(null), set TumblThree's proxy as PROXY: can open auth page (quick, which is odd since no record in PROXY'LOG, but always have login-error when logging, needs trying for servel times), can't add blog (all) and have no connect record in PROXY'LOG
unset IE's proxy, set TumblThree's proxy as **A-NOT-EXIST-PORT**(e.g. 127.0.0.1:1111, which has no app listening at it): can open auth page (which is even odder), can't add blog (all)
unset IE's proxy, set TumblThree's proxy as null: can open auth page (but take a long time, once loaded login is a one-time-success), can't add blog (all)

https://github.com/Emphasia/TumblThree-zh/issues/1
@1458271 uses another proxy tool(Shadowsocks@127.0.0.1:1080) and have the same problem.

In conclusion, the proxy setting in TumblThree have no practical effect. (but it used to be working fine.)
I made a wrong conclusion before and I’m trying to find out the version that started this bug, SORRY.

btw, maybe you can add error message to notify why the adding failed for reference?

**Comment by Sansming** [comment]
I also encountered the same problem
bug versions: 1.0.8.36, 1.0.8.39
normal version: 1.0.8.32
bug caused the problem: can not add a new blog

**Comment by johanneszab** [comment]
Thanks. I'll probably look into this and some other issues over the weekend.

**Comment by johanneszab** [comment]
I'm not sure how to quickly test this properly, so can anyone of you test if this release actually works behind a proxy?

You'll have to set the proxy in the windows (internet explorer) settings, TumblThree should use that.

[TumblThree-v1.0.8.40-rc-Application.zip](https://github.com/johanneszab/TumblThree/files/1754324/TumblThree-v1.0.8.40-rc-Application.zip)

**Comment by lovehn9527** [comment]
> 
> TumblThree-v1.0.8.40-rc-Application.zip

now. i tested it, it's looks normal!

![default](https://user-images.githubusercontent.com/10324170/36631222-3bbeb9e4-19af-11e8-9126-daadecba99b3.JPG)

**Comment by Emphasia** [comment]
@johanneszab yes, v1.0.8.40-rc's proxy works fine.

**Comment by johanneszab** [comment]
Thanks. I'll update the release at some point this weekend.

**Comment by johanneszab** [comment]
One last question.

Is it enough to set the windows (Internet explorer) proxy settings with the v1.0.8.41+ release? Then I will remove the proxy settings in the Settings window. Looks like people get more confused then necessary with two settings.

I'd really fix this for good since I see the issue there in China. But I cannot really test the code for myself, even after googling and trying to figure out how to cripple my connection. Thus programming is properly is really cumbersome.

Thanks for any feedback!

**Comment by Emphasia** [comment]
I think for now the proxy settings in the Settings window should be removed to avoid confusion.

> btw, maybe you can add error message to notify why the adding failed for reference?

but maybe add some tip on *current network-status* or error-message when *add-blog fail*?
e.g. "using IE's proxy setting now." "fail to add blog: connection fail"

**Comment by johanneszab** [comment]
So it works if you set the Windows proxy settings, and put nothing (empty box) in TumblThree?

**Comment by Emphasia** [comment]
@johanneszab yep.

**Comment by johanneszab** [comment]
Thanks!

I'll remove the settings then with the next release and update the readme accordingly.
